Migrated global-elements.adoc, added screenshots. (#1145)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/global-elements.adoc

= Global Elements
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:keywords: studio, server, components, connectors, elements, palette, global elements, configuration elements

> A Global Element is a reusable object containing parameters that any number of elements in a flow can share. You reference a global element from Anypoint Connectors or components in your Mule application.

Reviewer: Should this be presented as Global Element or global element? It is shown two different ways here, and throughout the Mule doc.

Use the global element to apply configuration details to multiple local elements in flows. Create one global element that defines parameters and configuration details, and then reference the global element from any flow element that uses this configuration. This practice enables you to ensure consistency across flow elements.

> Note that some components require their properties stored in a global element.

Reviewer: Do we need to say more about this? Is this covered in another topic?

> == Prerequisites

Reviewer: Prerequisites to what? Understanding GEs, or creating GEs? 
I think we typically use this title in the context of a Task, which doesn't show up here until later.

> Review the features of the xref:studio::index.adoc[Anypoint Studio Visual Editor] to learn more about developing Mule applications using the Anypoint Studio graphical user interface and to understand the context in which you use global elements.

Reviewer: ```suggestion
Review the features of the xref:studio::index.adoc[Anypoint Studio visual editor] to learn more about developing Mule applications using the Anypoint Studio graphical user interface and to understand the context in which you use global elements.
```

[[advantages]]
== Benefits

> Global elements provide many benefits to you as a Mule application developer:

Reviewer: ```suggestion
Global elements provide many benefits to you as a Mule application developer: reusability, efficiency, and flexibility.
```
OR
```suggestion
Global elements provide you, as a Mule application developer, with reusability, efficiency, and flexibility.
```

* Reusability
+

> You can configure a global element once, and then share that configuration among processors of the same type. Configuring global elements is particularly useful when you have to define the connection details and login credentials for an external source, such as a SaaS application or database.

Reviewer: ```suggestion
You can configure a global element once and then share that configuration among processors of the same type. Configuring global elements is particularly useful when you have to define the connection details and login credentials for an external source, such as a SaaS application or database.
```

* Efficiency
+

> You don't have to change the same configuration multiple times for different processors in the flow if they all share a global element. For example, if you have multiple connectors that connect to the same service using one set of credentials, you can change the configuration in the global element, and the changes apply to all flow elements that reference the global element.

Reviewer: ```suggestion
You don't have to change the same configuration multiple times for different processors in the flow if they all share a global element. For example, if you have multiple connectors that connect to the same service using one set of credentials, you can change the configuration in the global element and the changes apply to all flow elements that reference the global element.
```

* Flexibility
+
An element in a flow can apply a global element’s configuration exactly as defined, or it can apply modifications to the global configuration properties.

== Create a Global Element

You can create global elements in two ways:

* From the *Global Elements* tab in the Anypoint Studio canvas

> * From the properties panel of any Connector or Module that supports global elements

Reviewer: ```suggestion
* From the properties panel of any connector or module that supports global elements
```
Capitalize the names of connectors and modules, but not "connector" when used alone, and not "module" ever.

=== From the Anypoint Studio Canvas

> . In the Anypoint Studio Visual Editor, click the *Global Elements* tab to access a list of all global elements in an application.

Reviewer: ```suggestion
. In the Anypoint Studio visual editor, click the *Global Elements* tab to access a list of all global elements in an application:
```

+
image::mruntime-new-global-element.png[new-global-element]
+
. Click *Create* to add a new global element.

> . In the *Choose Global Type* wizard, navigate the directories or use the filter to select the type of global element you wish to create, then click *OK*.

Reviewer: ```suggestion
. In the *Choose Global Type* wizard, navigate the directories or use the filter to select the type of global element you want to create, and then click *OK*.
```

> . Define the configurable parameters of your global element in the *Global Element Properties* window that shows.

Reviewer: ```suggestion
. Define the configurable parameters of your global element in the *Global Element Properties* window.
```

. Click *OK* to save.

=== From the Properties Panel of a Connector or Module

To create a new global element from within a local component you placed in your flow:

. Click the component you want to create a global element for:
+
image::mruntime-global-element-component.png[select-component-global-element]
+

> . In the properties panel, click the Add icon next to the *Connector Configuration*.

Reviewer: ```suggestion
. In the properties panel, click the Add icon next to the *Connector Configuration*:
```

+
image::mruntime-global-element-create.png[create-config-global-element]

> . Fill in the required fields for the type of global element you want to create.

Reviewer: ```suggestion
. Complete the required fields for the type of global element you want to create.
```
